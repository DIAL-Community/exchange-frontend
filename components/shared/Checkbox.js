export const Checkbox = ({ type, value, onChange, onBlur, className = '', ...otherProps }) => (
  <input
    {...otherProps}
    type='checkbox'
    value={value}
    onChange={onChange}
    onBlur={onBlur}
    className={`${className} h-8 w-8`}
    data-testid='checkbox'
  />
)
