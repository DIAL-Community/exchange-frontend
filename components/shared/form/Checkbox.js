import React from 'react'
import classNames from 'classnames'

const Checkbox = React.forwardRef(({ value, onChange, onBlur, className, ...otherProps }, ref) => (
  <input
    {...otherProps}
    ref={ref}
    type='checkbox'
    checked={value}
    onChange={onChange}
    onBlur={onBlur}
    className={classNames(className, 'h-5 w-5')}
  />
))

Checkbox.displayName = 'Checkbox'

export default Checkbox
