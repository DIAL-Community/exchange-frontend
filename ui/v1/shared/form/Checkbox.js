import React from 'react'
import classNames from 'classnames'

const Checkbox = React.forwardRef(({ value, onChange, onBlur, className, ...otherProps }, ref) => (
  <input
    {...otherProps}
    ref={ref}
    type='checkbox'
    checked={value}
    onChange={onChange}
    onBlur={onBlur}
    className={classNames(className, 'h-6 w-6')}
    data-testid='checkbox'
  />
))

Checkbox.displayName = 'Checkbox'

export default Checkbox
