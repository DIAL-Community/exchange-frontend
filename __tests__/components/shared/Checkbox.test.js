import { render, fireEvent } from '@testing-library/react'
import Checkbox from '../../../components/shared/Checkbox'

describe('Unit test for the Checkbox component.', () => {
  const TEST_ID = 'checkbox'

  test('Should match snapshot - unchecked.', () => {
    const onChange = jest.fn()
    const input = render(<Checkbox value={false} onChange={onChange} />).getByTestId(TEST_ID)
    expect(input).toMatchSnapshot()
    expect(input).not.toBeChecked()
  })

  test('Should match snapshot - checked.', () => {
    const onChange = jest.fn()
    const input = render(<Checkbox value={true} onChange={onChange} />).getByTestId(TEST_ID)
    expect(input).toMatchSnapshot()
    expect(input).toBeChecked()
  })

  test('Should call onChange function.', () => {
    const onChange = jest.fn()
    const input = render(<Checkbox value={false} onChange={onChange} />).getByTestId(TEST_ID)
    fireEvent.click(input)
    expect(onChange).toBeCalledTimes(1)
  })
})
