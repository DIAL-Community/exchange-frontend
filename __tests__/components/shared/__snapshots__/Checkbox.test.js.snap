// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the Checkbox component. Should match snapshot - checked. 1`] = `
<input
  checked=""
  class="h-8 w-8"
  data-testid="checkbox"
  type="checkbox"
/>
`;

exports[`Unit test for the Checkbox component. Should match snapshot - unchecked. 1`] = `
<input
  class="h-8 w-8"
  data-testid="checkbox"
  type="checkbox"
/>
`;
